Server/src/services/vehicle.service.ts

```diff
@@ -140,26 +124,27 @@ export default class VehicleService {
 				return null
 			}
 
-			allVehiclesOnTrack.filter(async function (vehicle, _index, _vehicles) {
-				const vehicleTrackKm = await VehicleService.getVehicleTrackDistanceKm(vehicle, track)
-				if (vehicleTrackKm == null) {
-					// TODO: log this
-					return null
-				}
-				return vehicleTrackKm - trackDistance * heading > 0
-			})
-		}
-
-		// filter vehicles by distance if given
-		if (maxDistance != null) {
-			allVehiclesOnTrack.filter(async function (vehicle, _index, _vehicles) {
-				const vehicleTrackKm = await VehicleService.getVehicleTrackDistanceKm(vehicle, track)
-				if (vehicleTrackKm == null) {
-					return false
-				}
-				return Math.abs(vehicleTrackKm - trackDistance) < maxDistance
-			})
-		}
+            allVehiclesOnTrack.filter(async function (vehicle, index, vehicles){
+                const vehicleTrackKm = await VehicleService.getVehicleTrackDistanceKm(vehicle)
+                if (vehicleTrackKm == null) {
```

Comment:
Not an error but more like an inconsistency:
Sometimes you use the `??` (Nullish Coalescing) operator to either return the "right" value or if it is "null" another value, and then sometimes you use a if statement.